.github/workflows/test-python-ci.yml

```diff
@@ -0,0 +1,121 @@
+name: test-python-ci
+
+on:
+  pull_request:
+    paths:
+      - parsec/**
+      - tests/**
+      - pyproject.toml
+      - poetry.lock
+      - .github/workflows/test-python-ci.yml
```

Comment:
Don't we want to run this if anything in `.github/` is modified ? (typically in case the .github/actions is modified)

Reply:
It's a choice to be made

If we want to, since we will be required to also change `test-rust-ci`, I think that's also for another PR